File: src/libs/pdflib/bind/pdflib/java/pdfclock.java
/* $Id: pdfclock.java 14574 2005-10-29 16:27:43Z bonefish $
 *
 * PDFlib client: pdfclock example in Java
 */

import java.io.*;
import java.text.*;		// SimpleDateFormat
import java.util.*;		// Date
import com.pdflib.pdflib;
import com.pdflib.PDFlibException;

public class pdfclock
{
    public static void main (String argv[])
    {
	pdflib p = null;
	int tm_hour, tm_min, tm_sec, alpha;
	float RADIUS = 200.0f;
	float MARGIN = 20.0f;
	SimpleDateFormat format;
	Date now = new Date();

	try{
	    p = new pdflib();
	    /* open new PDF file */
	    if (p.open_file("pdfclock.pdf") == -1) {
		throw new Exception("Error: " + p.get_errmsg());
	    }

	    p.set_info("Creator", "pdfclock.java");
	    p.set_info("Author", "Thomas Merz");
	    p.set_info("Title", "PDF clock (Java)");

	    p.begin_page(  (int) (2 * (RADIUS + MARGIN)),
				    (int) (2 * (RADIUS + MARGIN)));

	    p.translate(RADIUS + MARGIN, RADIUS + MARGIN);
	    p.setcolor("fillstroke", "rgb", 0.0f, 0.0f, 1.0f, 0.0f);
	    p.save();

	    // minute strokes 
	    p.setlinewidth(2.0f);
	    for (alpha = 0; alpha < 360; alpha += 6)
	    {
		p.rotate(6.0f);
		p.moveto(RADIUS, 0.0f);
		p.lineto(RADIUS-MARGIN/3, 0.0f);
		p.stroke();
	    }

	    p.restore();
	    p.save();

	    // 5 minute strokes
	    p.setlinewidth(3.0f);
	    for (alpha = 0; alpha < 360; alpha += 30)
	    {
		p.rotate(30.0f);
		p.moveto(RADIUS, 0.0f);
		p.lineto(RADIUS-MARGIN, 0.0f);
		p.stroke();
	    }

	    format = new SimpleDateFormat("hh");
	    tm_hour= Integer.parseInt(format.format(now));
	    format = new SimpleDateFormat("mm");
	    tm_min = Integer.parseInt(format.format(now));
	    format = new SimpleDateFormat("ss");
	    tm_sec = Integer.parseInt(format.format(now));

	    // draw hour hand 
	    p.save();
	    p.rotate((-((tm_min/60.0f) + tm_hour - 3.0f) * 30.0f));
	    p.moveto(-RADIUS/10, -RADIUS/20);
	    p.lineto(RADIUS/2, 0.0f);
	    p.lineto(-RADIUS/10, RADIUS/20);
	    p.closepath();
	    p.fill();
	    p.restore();

	    // draw minute hand
	    p.save();
	    p.rotate((-((tm_sec/60.0f) + tm_min - 15.0f) * 6.0f));
	    p.moveto(-RADIUS/10, -RADIUS/20);
	    p.lineto(RADIUS * 0.8f, 0.0f);
	    p.lineto(-RADIUS/10, RADIUS/20);
	    p.closepath();
	    p.fill();
	    p.restore();

	    // draw second hand
	    p.setcolor("fillstroke", "rgb", 1.0f, 0.0f, 0.0f, 0.0f);
	    p.setlinewidth(2);
	    p.save();
	    p.rotate(-((tm_sec - 15.0f) * 6.0f));
	    p.moveto(-RADIUS/5, 0.0f);
	    p.lineto(RADIUS, 0.0f);
	    p.stroke();
	    p.restore();

	    // draw little circle at center
	    p.circle(0f, 0f, RADIUS/30);
	    p.fill();

	    p.restore();
	    p.end_page();
	    p.close();

        } catch (PDFlibException e) {
	    System.err.print("PDFlib exception occurred in pdfclock sample:\n");
	    System.err.print("[" + e.get_errnum() + "] " + e.get_apiname() +
			    ": " + e.getMessage() + "\n");
        } catch (Exception e) {
            System.err.println(e.getMessage());
        } finally {
            if (p != null) {
		p.delete();			/* delete the PDFlib object */
            }
        }
    }
}

File: src/libs/pdflib/bind/pdflib/php/pdfclock.php
<?php
/* $Id: pdfclock.php 14574 2005-10-29 16:27:43Z bonefish $
 * A little PDFlib application to draw an analog clock.
 *
 */
$RADIUS = 200.0;
$MARGIN = 20.0;

$p = PDF_new();					/* create a new PDFlib object */

/*  open new PDF file; insert a file name to create the PDF on disk */
if (PDF_open_file($p, "") == 0) {
    die("Error: " . PDF_get_errmsg($p));
}

/* This line is required to avoid problems on Japanese systems */
PDF_set_parameter($p, "hypertextencoding", "winansi");

PDF_set_info($p, "Creator", "pdfclock.php");
PDF_set_info($p, "Author", "Rainer Schaaf");
PDF_set_info($p, "Title", "PDF clock (PHP)");
						/* start a new page     */
PDF_begin_page($p, 2 * ($RADIUS + $MARGIN), 2 * ($RADIUS + $MARGIN));

PDF_translate($p, $RADIUS + $MARGIN, $RADIUS + $MARGIN);
PDF_setcolor($p, "fillstroke", "rgb", 0.0, 0.0, 1.0, 0.0);
PDF_save($p);

/* minute strokes */
PDF_setlinewidth($p, 2.0);
for ($alpha = 0; $alpha < 360; $alpha += 6)
{
    PDF_rotate($p, 6.0);
    PDF_moveto($p, $RADIUS, 0.0);
    PDF_lineto($p, $RADIUS-$MARGIN/3, 0.0);
    PDF_stroke($p);
}

PDF_restore($p);
PDF_save($p);

/* 5 minute strokes */
PDF_setlinewidth($p, 3.0);
for ($alpha = 0; $alpha < 360; $alpha += 30)
{
    PDF_rotate($p, 30.0);
    PDF_moveto($p, $RADIUS, 0.0);
    PDF_lineto($p, $RADIUS-$MARGIN, 0.0);
    PDF_stroke($p);
}

$ltime = getdate();

/* draw hour hand */
PDF_save($p);
PDF_rotate($p, -(($ltime['minutes']/60.0)+$ltime['hours']-3.0)*30.0);
PDF_moveto($p, -$RADIUS/10, -$RADIUS/20);
PDF_lineto($p, $RADIUS/2, 0.0);
PDF_lineto($p, -$RADIUS/10, $RADIUS/20);
PDF_closepath($p);
PDF_fill($p);
PDF_restore($p);

/* draw minute hand */
PDF_save($p);
PDF_rotate($p, -(($ltime['seconds']/60.0)+$ltime['minutes']-15.0)*6.0);
PDF_moveto($p, -$RADIUS/10, -$RADIUS/20);
PDF_lineto($p, $RADIUS * 0.8, 0.0);
PDF_lineto($p, -$RADIUS/10, $RADIUS/20);
PDF_closepath($p);
PDF_fill($p);
PDF_restore($p);

/* draw second hand */
PDF_setcolor($p, "fillstroke", "rgb", 1.0, 0.0, 0.0, 0.0);
PDF_setlinewidth($p, 2);
PDF_save($p);
PDF_rotate($p, -(($ltime['seconds'] - 15.0) * 6.0));
PDF_moveto($p, -$RADIUS/5, 0.0);
PDF_lineto($p, $RADIUS, 0.0);
PDF_stroke($p);
PDF_restore($p);

/* draw little circle at center */
PDF_circle($p, 0, 0, $RADIUS/30);
PDF_fill($p);

PDF_restore($p);
PDF_end_page($p);				/* close page           */

PDF_close($p);					/* close PDF document	*/

$buf = PDF_get_buffer($p);
$len = strlen($buf);

header("Content-type: application/pdf");
header("Content-Length: $len");
header("Content-Disposition: inline; filename=pdfclock.pdf");
print $buf;

PDF_delete($p);					/* delete the PDFlib object */
?>
